src/mapping/AxialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,148 @@
+#include "AxialGeoMultiscaleMapping.hpp"
+#include <Eigen/src/Core/Matrix.h>
+#include "mesh/Mesh.hpp"
+
+namespace precice::mapping {
+
+AxialGeoMultiscaleMapping::AxialGeoMultiscaleMapping(
+    Constraint     constraint,
+    int            dimensions,
+    MultiscaleType type,
+    MultiscaleAxis axis,
+    double         radius)
+    : Mapping(constraint, dimensions, false, Mapping::InitialGuessRequirement::None),
+      _type(type),
+      _axis(axis),
+      _radius(radius)
+{
+  setInputRequirement(Mapping::MeshRequirement::VERTEX);
+  setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+}
+
+void AxialGeoMultiscaleMapping::computeMapping()
+{
+  PRECICE_TRACE(output()->vertices().size());
+
+  PRECICE_ASSERT(input().get() != nullptr);
+  PRECICE_ASSERT(output().get() != nullptr);
+
+  if (getConstraint() == CONSISTENT) {
+    PRECICE_DEBUG("Compute consistent mapping");
+    if (_type == SPREAD) {
+      PRECICE_CHECK(input()->vertices().size() == 1, "You can only define an axial geometric multiscale mapping of type spread from a mesh with exactly one vertex.");
+    } else {
+      PRECICE_ASSERT(_type == COLLECT);
+      PRECICE_CHECK(output()->vertices().size() == 1, "You can only define an axial geometric multiscale mapping of type spread from a mesh with exactly one vertex.");
+    }
+  } else {
+    PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+    PRECICE_ASSERT(false, "Not yet implemented");
+    PRECICE_DEBUG("Compute conservative mapping");
+  }
+  _hasComputedMapping = true;
+}
+
+void AxialGeoMultiscaleMapping::clear()
+{
+  PRECICE_TRACE();
+  _hasComputedMapping = false;
+}
+
+void AxialGeoMultiscaleMapping::mapConservative(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+  PRECICE_ASSERT(false, "Not yet implemented");
+  PRECICE_DEBUG("Map conservative");
+}
+
+void AxialGeoMultiscaleMapping::mapConsistent(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_TRACE();
+
+  const int              inValueDimensions = inData.dataDims;
+  const Eigen::VectorXd &inputValues       = inData.values;
+  Eigen::VectorXd &      outputValues      = outData;
+  // TODO: check how to access correct outValueDimensions
+  const int outValueDimensions = outData.size() / output()->vertices().size();
```

Comment:
We need to check whether the way we compute the outValueDimensions is the optimal way. It should be correct as the tests pass, but we do not have a simple `dataDims` member that we can call. @fsimonis do you have any thoughts on that?

Reply:
In the meantime, the dimensions attribute was moved from the `solver-interface` to each mesh. However, we don't yet have support for 1D meshes. The current way seems fine to me, we anyway need to update it once we have 1D meshes.